src/llama.cpp

```diff
@@ -2661,6 +2664,9 @@ struct llama_context {
     struct ggml_tensor * inp_s_mask;    // F32 [1, n_kv]
     struct ggml_tensor * inp_s_seq;     // I32 [n_kv, n_batch]
 
+    // KQ mask per layer, used by sliding window attention (gemma 2)
+    std::vector<struct ggml_tensor *> inp_KQ_mask_l;
```

Comment:
As far as I can tell there are only two KQ masks now and this is not necessary. It would be cleaner with a single `inp_KQ_mask_SWA` and a corresponding `build_inp_KQ_mask_SWA` function.

Reply:
I agree, the `std::vector` is quite future-proof and unnecessary atm. I'm a bit bother duplicating `build_inp_KQ_mask` function, so ended up adding a param for it. Do you think it's ok now? https://github.com/ggerganov/llama.cpp/pull/8227/commits/d09ecb84c8300504bb76794298c0ef47d541a733

(I tested on gemma 2 + llama 3 to confirm that it doesn't break anything)